### DISCUSSION TITLE
Problems connecting to bridge fw version 3.8.4

### DISCUSSION THREAD

**Discussion by rajohn96** [aarons22/homebridge-bond#240] [General]
All,

I have a home bridge with fw v3.8.4 that I can see via ping and curl to without issue but cannot connect via the homebridge plug-in. I get the following message in the log:

[Bond] Unable to find Bond for IP Address: 192.168.1.36. Skipping this Bond.

Even if I restart the bridge and then restart the homebridge app within the 10 minute window (during which time no token is required) I have the same results. This seems to be something with the axios library throwing an ECONNABORTED error.

TIA!

**Comment by barjam** [comment]
Hi rajohn96. I am having the same issue. It looks like it tries to run an initial "ping" command hitting http://${this.bondIP}/v2/devices and fails. I can hit this address with CURL though so not sure what the problem is. I plan to mess with the code later today to see if I can figure anything out.

**Comment by barjam** [comment]
I set Home Bridge to -D and things started working again even when turning of -D. No idea why. 

Edit: Turning -D back off didn't seem reliable. I set bond to it's own bridge mode and that seems to work every time. I tried isolating the code locally for ping and that worked every time. I think *maybe* what is happening is HAP (whatever that is) fires up but isn't quite done when bond tries to do its thing? I see the plugin uses HAP and in my logs HAP starting is mentioned right before bond. Everything is working for me now so I don't play to dig into this any more.

**Reply by rajohn96** [reply]
yes, mine worked when I turned on the -D but then stopped working with I turned it off. If it is a timing thing then turning on the -D (which slows things down) likely could be the trick (not a good one mind you)

**Reply by rajohn96** [reply]
well i say it worked, but now i'm not so sure; not throwing the Unable to find error, but also not able to control the devices, while the app can just fine. grrr ....

**Reply by barjam** [reply]
Try setting the bond plugin in "bridge" mode. You have to register it separately in HomeKit but seems to be working fine for me.

**Reply by rajohn96** [reply]
OK, did that and now it's not throwing errors in Homebridge, but neither it (via Accessories) nor Homekit are seeing the devices on the Bond, but the Bond App does. Are there other steps here? For my TPLink it was very straightforward, and I see the devices in both as expected. Do I still need to be in debug mode?

**Reply by rajohn96** [reply]
Never mind; i wasn't patient enough :-) Seems to be working now, and no debug mode!! Thanks for the assist @barjam; you rock!

**Reply by droopdog7** [reply]
What is "-D" exactly?

**Comment by eburganjr** [comment]
THANK YOU both for this. I was banging my head against the wall trying to figure this out. Ultimately, I also had to put Bond in Bridge Mode and THEN restart the child bridge to get it to connect. I guess it is a timing thing with HAP or whatever that is (my technical knowledge is somewhat lacking there).
